embassy-mcxa/src/reset_reason.rs

```diff
@@ -0,0 +1,108 @@
+//! Reset reason
+//!
+//! MCXA families keep the most recent reset reason in the SRS
+//! register of the CMC block. This lets users understand why the MCU
+//! has reset and take appropriate corrective actions if required.
+
+/// Reads the most recent reset reason from the Core Mode Controller
+/// (CMC).
+pub fn reset_reason() -> ResetReason {
+    critical_section::with(|_| {
+        let regs = unsafe { &*crate::pac::Cmc::steal() };
+
+        let srs = regs.srs().read();
+
+        if srs.wakeup().is_enabled() {
```

Comment:
I'm not sure if more than one item can be set, but if so, we might want something like we have for DMA errors: https://github.com/embassy-rs/embassy/blob/32a1d0ef7ea52d3cee9959ff52d47fd13fc6b4b9/embassy-mcxa/src/dma.rs#L1836-L1965